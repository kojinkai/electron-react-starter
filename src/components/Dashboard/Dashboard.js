import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Dashboard.css';

export class Dashboard extends Component {
  render() {
    return (
      <div className="dashboard">
        <div className="dashboard__items">
          {this.props.dashboardItems.map(item =>
            <div key={item.id} className="dashboard__item">
              <span className="dashboard__item-inner">
                {item.name}
              </span>
            </div>
          )}
        </div>
        <button className="dashboard__add-icon" onClick={
          () => this.props.saveDashboardItem({
            name: `item ${this.props.dashboardItems.length + 1}`,
          })
        }>
          <span className="dashboard__add-icon-bar dashboard__add-icon-bar--horizontal"></span>
          <span className="dashboard__add-icon-bar dashboard__add-icon-bar--vertical"></span>
        </button>
      </div>
    );
  }
}

Dashboard.propTypes = {
  dashboardItems: PropTypes.array.isRequired,
  saveDashboardItem: PropTypes.func.isRequired,
};

export default Dashboard;
