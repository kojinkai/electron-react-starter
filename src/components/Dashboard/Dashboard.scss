@import 'styles/grid';

.dashboard {
  padding: $gutter;
}