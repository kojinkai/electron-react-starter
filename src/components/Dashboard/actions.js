import { createAction } from 'redux-actions';
export const addDashboardItem = createAction('ADD_DASHBOARD_ITEM');
