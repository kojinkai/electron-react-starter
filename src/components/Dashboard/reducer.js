// import { fromJS } from 'immutable';
import { handleActions } from 'redux-actions';

const dashBoardReducer = handleActions({}, null);

export default dashBoardReducer;
