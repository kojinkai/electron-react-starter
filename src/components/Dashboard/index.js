import { connect } from 'react-redux';
import Dashboard from './Dashboard';

const mapStateToProps = state => console.log('state is: ', state) || ({
  dashboardItems: state.dashboard.toJS(),
});

export default connect(mapStateToProps)(Dashboard);
