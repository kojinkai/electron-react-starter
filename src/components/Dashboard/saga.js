import { call, put, takeEvery } from 'redux-saga/effects'
import { saveDashboardItem, completeSaveDashboardItem, failedSaveDashboardItem } from './actions';
import { saveToDB } from '../../db/configureIDB';
import { getActionType } from '../../utils';

export function* saveItem(action) {
   try {
      const saveRequest = yield call(saveToDB, action.payload);
      yield put(completeSaveDashboardItem(saveRequest));
   } catch (e) {
      yield put(failedSaveDashboardItem());
   }
}

export function* takeSaveDashboardItem() {
  yield takeEvery(getActionType(saveDashboardItem), saveItem);
}

export default takeSaveDashboardItem;
