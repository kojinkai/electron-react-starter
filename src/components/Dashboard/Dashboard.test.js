import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from './Dashboard';
const mockItems = [{ id: 1234, name: 'foo' }, { id: 2345, name: 'bar' }];

it('renders according to the snapshot', () => {
  const wrapper = shallow(
    <Dashboard
      dashboardItems={mockItems}
      saveDashboardItem={jest.fn()}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should call the saveDashboardItem prop with a new item when the add button is clicked', () => {
  const saveDashboardItemSpy = jest.fn();
  const wrapper = shallow(
    <Dashboard
      dashboardItems={mockItems}
      saveDashboardItem={saveDashboardItemSpy}
    />
  );

  wrapper.find('button').simulate('click');

  expect(saveDashboardItemSpy).toHaveBeenCalledWith({
    name: 'item 3',
  });
});
