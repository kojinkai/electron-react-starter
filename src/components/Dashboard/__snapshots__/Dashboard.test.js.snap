// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders according to the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    dashboardItems={
        Array [
            Object {
              "id": 1234,
              "name": "foo",
            },
            Object {
              "id": 2345,
              "name": "bar",
            },
          ]
    }
    saveDashboardItem={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="dashboard__items"
>
          <div
                    className="dashboard__item"
          >
                    <span
                              className="dashboard__item-inner"
                    >
                              foo
                    </span>
          </div>
          <div
                    className="dashboard__item"
          >
                    <span
                              className="dashboard__item-inner"
                    >
                              bar
                    </span>
          </div>
</div>,
        <button
          className="dashboard__add-icon"
          onClick={[Function]}
>
          <span
                    className="dashboard__add-icon-bar dashboard__add-icon-bar--horizontal"
          />
          <span
                    className="dashboard__add-icon-bar dashboard__add-icon-bar--vertical"
          />
</button>,
      ],
      "className": "dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="dashboard__item"
>
              <span
                            className="dashboard__item-inner"
              >
                            foo
              </span>
</div>,
            <div
              className="dashboard__item"
>
              <span
                            className="dashboard__item-inner"
              >
                            bar
              </span>
</div>,
          ],
          "className": "dashboard__items",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1234",
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="dashboard__item-inner"
>
                foo
</span>,
              "className": "dashboard__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "foo",
                "className": "dashboard__item-inner",
              },
              "ref": null,
              "rendered": "foo",
              "type": "span",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "2345",
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="dashboard__item-inner"
>
                bar
</span>,
              "className": "dashboard__item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "bar",
                "className": "dashboard__item-inner",
              },
              "ref": null,
              "rendered": "bar",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="dashboard__add-icon-bar dashboard__add-icon-bar--horizontal"
/>,
            <span
              className="dashboard__add-icon-bar dashboard__add-icon-bar--vertical"
/>,
          ],
          "className": "dashboard__add-icon",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dashboard__add-icon-bar dashboard__add-icon-bar--horizontal",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "dashboard__add-icon-bar dashboard__add-icon-bar--vertical",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="dashboard__items"
>
            <div
                        className="dashboard__item"
            >
                        <span
                                    className="dashboard__item-inner"
                        >
                                    foo
                        </span>
            </div>
            <div
                        className="dashboard__item"
            >
                        <span
                                    className="dashboard__item-inner"
                        >
                                    bar
                        </span>
            </div>
</div>,
          <button
            className="dashboard__add-icon"
            onClick={[Function]}
>
            <span
                        className="dashboard__add-icon-bar dashboard__add-icon-bar--horizontal"
            />
            <span
                        className="dashboard__add-icon-bar dashboard__add-icon-bar--vertical"
            />
</button>,
        ],
        "className": "dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="dashboard__item"
>
                <span
                                className="dashboard__item-inner"
                >
                                foo
                </span>
</div>,
              <div
                className="dashboard__item"
>
                <span
                                className="dashboard__item-inner"
                >
                                bar
                </span>
</div>,
            ],
            "className": "dashboard__items",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1234",
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="dashboard__item-inner"
>
                  foo
</span>,
                "className": "dashboard__item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "foo",
                  "className": "dashboard__item-inner",
                },
                "ref": null,
                "rendered": "foo",
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "2345",
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="dashboard__item-inner"
>
                  bar
</span>,
                "className": "dashboard__item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "bar",
                  "className": "dashboard__item-inner",
                },
                "ref": null,
                "rendered": "bar",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="dashboard__add-icon-bar dashboard__add-icon-bar--horizontal"
/>,
              <span
                className="dashboard__add-icon-bar dashboard__add-icon-bar--vertical"
/>,
            ],
            "className": "dashboard__add-icon",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dashboard__add-icon-bar dashboard__add-icon-bar--horizontal",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "dashboard__add-icon-bar dashboard__add-icon-bar--vertical",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
